### PR TITLE
fixes #7989 - API v2 - support unwrapped parameters when doing POST/PUT to nested routes

### PR DESCRIPTION
**Pull request by shlomizadok**
added operating_id parameter to hosts and auth_source_id parameter to users in api/v2

### CODE REVIEW

**Comment by domcleal** [comment]
This fix looks incorrect to me, those are attributes of the host or user, but this method is a whitelist of IDs that may be used in nested URLs.

**Comment by shlomizadok** [comment]
@domcleal - I agree, rushed into a solution without understanding how it works. Re-did another wider solution

**Comment by isratrade** [comment]
@shlomizadok, looks good, but I would change the commit message to reflect the fix.

**Comment by shlomizadok** [comment]
@isratrade -- changed.

File: app/controllers/api/base_controller.rb [hunk 1]
```diff
@@ -189,17 +189,28 @@ def log_response_body
     attr_reader :nested_obj
 
     def find_required_nested_object
```
Comment (dLobatog) [inline app/controllers/api/base_controller.rb:191]: `find_required_nested_object` and ````find_optional_required_nested_object` share 4/5 lines, please DRY it up

File: app/controllers/api/base_controller.rb [hunk 2]
```diff
@@ -189,17 +189,28 @@ def log_response_body
...
+      return if is_flat?(params) && actions.include?(action_name)
       find_nested_object
       return @nested_obj if @nested_obj
       not_found
     end
 
     def find_optional_nested_object
+      actions = %w(create update)
+      return if is_flat?(params) && actions.include?(action_name)
+      find_nested_object
```
Comment (dLobatog) [inline app/controllers/api/base_controller.rb:202]: No need to `find_nested_object` twice, probably just a typo.

File: app/controllers/api/base_controller.rb (hunk 1 above)
Comment (shlomizadok) [inline app/controllers/api/base_controller.rb:191]: DRYed it up. Thanks.

**Comment by dLobatog** [comment]
:clap: Works great, merging in 3..2..1!

**Comment by dLobatog** [comment]
Merged as 47ab039e61ae3f29627de0a770fa0c6b41f33c47 , thanks @shlomizadok :wine_glass:

**Pull request closed without merging**